cmake/FindWin32SslRuntime.cmake

```diff
@@ -1,66 +1,47 @@
 # Find the OpenSSL runtime libraries (.dll) for Windows that
 # will be needed by Qt in order to access https urls.
-
 if (WIN32)
-  # Get standard installation paths for OpenSSL under Windows
-
-  # http://www.slproweb.com/products/Win32OpenSSL.html
-
-  if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-    # target win64
-    set(_OPENSSL_ROOT_HINTS
-      ${OPENSSL_ROOT_DIR}
-      "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\OpenSSL (64-bit)_is1;Inno Setup: App Path]"
-      ENV OPENSSL_ROOT_DIR
-      )
+  if("${CMAKE_GENERATOR_PLATFORM}" STREQUAL "x64")
+    message(STATUS "Looking for OpenSSL for ${CMAKE_GENERATOR_PLATFORM}")
+    set(_OPENSSL_ROOT_HINTS ${OPENSSL_ROOT_DIR} ENV OPENSSL_ROOT_DIR)
     file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _programfiles)
     set(_OPENSSL_ROOT_PATHS
-      "C:/Tools/vcpkg/installed/x64-windows/bin"
-      "${_programfiles}/OpenSSL-Win64"
-      "C:/OpenSSL-Win64/"
-      )
+            "$ENV{VCPKG_PACKAGES_DIR}/x64-windows/bin"
+            "C:/OpenSSL-Win64/bin"
+            "C:/OpenSSL-Win64"
+            "C:/Tools/vcpkg/installed/x64-windows/bin"
+            "${_programfiles}/OpenSSL-Win64"
+            )
     unset(_programfiles)
-  else( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-    # target win32
-    set(_OPENSSL_ROOT_HINTS
-      ${OPENSSL_ROOT_DIR}
-      "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\OpenSSL (32-bit)_is1;Inno Setup: App Path]"
-      ENV OPENSSL_ROOT_DIR
-      )
+  elseif("${CMAKE_GENERATOR_PLATFORM}" STREQUAL "Win32")
+    message(STATUS "Looking for OpenSSL for ${CMAKE_GENERATOR_PLATFORM}")
+    set(_OPENSSL_ROOT_HINTS ${OPENSSL_ROOT_DIR} ENV OPENSSL_ROOT_DIR)
     file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _programfiles)
     set(_OPENSSL_ROOT_PATHS
-      "C:/Tools/vcpkg/installed/x86-windows/bin"
-      "${_programfiles}/OpenSSL"
-      "${_programfiles}/OpenSSL-Win32"
-      "C:/OpenSSL/"
-      "C:/OpenSSL-Win32/"
-      )
+            "$ENV{VCPKG_PACKAGES_DIR}/x86-windows/bin"
+            "C:/OpenSSL-Win32/bin"
+            "C:/OpenSSL-Win32"
+            "C:/OpenSSL"
+            "C:/Tools/vcpkg/installed/x86-windows/bin"
+            "${_programfiles}/OpenSSL"
+            "${_programfiles}/OpenSSL-Win32"
+            )
     unset(_programfiles)
-  endif( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-
+  else()
+    message(WARNING "Unknown OS_PATH: $ENV{OS_PATH}, Can't find OpenSSL correctly")
+  endif()
 else ()
-  set(_OPENSSL_ROOT_HINTS
-    ${OPENSSL_ROOT_DIR}
-    ENV OPENSSL_ROOT_DIR
-    )
-endif ()
-
-set(_OPENSSL_ROOT_HINTS_AND_PATHS
```

Comment:
are you sure this part was unnecessary? you do seem to be setting the _ROOT_HINTS earlier

Reply:
I was also removing them from setting. HINTS and PATHS are both optional, I'm only using Paths rn for it, which is fine